### DISCUSSION TITLE
Typescript code splitting endpoints type

### DISCUSSION THREAD

**Discussion by Janaka-Steph** [reduxjs/redux-toolkit#2199] [Q&A] [answered]
Hello! I have an issue with Typescript in code splitting context.
```
export const baseApi = createApi({
  baseQuery: fakeBaseQuery<string>(),
  endpoints: () => ({}),
});
```
```
const walletApi = baseApi.injectEndpoints({
  endpoints: (build) => ({
    walletAddress: build.query<WalletAddressReply, void>({ ... })
  })
})  
```
```
baseApi.endpoints.walletAddress.initiate()
```
TS Error: property `walletAddress` does not exist on type '{}'.
This is because baseApi `endpoints` didn't receive the types of all injected apis. 

What is the proper way of fixing the `endpoints` type? Thank you

**Comment by Janaka-Steph** [comment, marked as answer]
Actually I suppose I have to export `walletApi` and use it like so `walletApi.endpoints.walletAddress.initiate()` instead of calling it from `baseApi`

**Reply by phryneas** [reply]
Exactly. That's also the only reliable way to make sure that the `injectEndpoints` has been called before you access the endpoint.
